core/domain/param_domain.py

```diff
@@ -205,15 +205,14 @@ def validate(self):
                 'Only parameter names with characters in [a-zA-Z0-9] are '
                 'accepted.')
 
-        try:
-            self.generator
-        except KeyError:
-            raise utils.ValidationError(
-                'Invalid generator id %s' % self._generator_id)
-        except Exception:
-            raise utils.ValidationError(
-                'Expected generator id to be a string, received %s '
-                % (self._generator_id))
+        if not hasattr(self, 'generator'):
+            try:
+                raise utils.ValidationError(
+                    'Invalid generator id %s' % self._generator_id)
+            except Exception:
```

Comment:
Instead of wrapping this under try-except, can we have an if the check for _generator_id to be a string? (Like the way we have  done for the Key Error)

Reply:
Done.